### PR TITLE
Generating /feed.xml with all EIPs

### PR DESCRIPTION
**Pull request by bmann**
Partially addresses #1688 

Global ```feed.xml``` that includes all EIPs. You can see a preview here https://5c3a52345d52e7000834b647--ethereum-eips-refactor.netlify.com/ --> and the direct link to feed.xml is here --> https://5c3a52345d52e7000834b647--ethereum-eips-refactor.netlify.com/feed.xml

[W3C Validator passed as valid RSS](https://validator.w3.org/feed/check.cgi?url=https%3A%2F%2F5c3a52345d52e7000834b647--ethereum-eips-refactor.netlify.com%2Ffeed.xml) -- with caveats about wrong link (since link is set to the live EIPS url), including author emails that look like tags, and use of relative URLs (which works in Atom).

Last Call metadata of ```review-end-date```  is done (previously merged in).

A couple of notes:
* refactored eipcontent generation into an include so all feeds can have the same internal content
* created is a manual data by the EIP author -- this is what pubDate is set to
* for the feed lastPubDate I am using the build date -- so every time the site is built, this field will change (since it usually gets built when EIPs are merged, this is OK)
* on Github Pages, you can't easily run a last-modified-date plugin -- we may want to encourage EIP authors to add / edit this manually as they majorly refactor Drafts
* to make more changes to the site -- like adding feeds to All/Core/Networking/etc -- the minima theme that is used needs to have files brought in; I brought in default.html for starters, which has /feed.xml hardcoded in the header by default

### CODE REVIEW

**Comment by bmann** [comment]
@fulldecent if you can review this and check the sample feeds here -- and tell me what RSS reader you use -- that would be helpful.

**Comment by fulldecent** [comment]
Thank you for sharing. I have perused the change and believe it does not resolve this issue:

https://github.com/ethereum/EIPs/issues/1688#issuecomment-453354562

Please update so it resolves this issue. Specifically, for the Last Call RSS, the GUID should include the date.

**Comment by bmann** [comment]
The GUID is unique for RSS feeds @fulldecent and shouldn’t change. 

I change the description on status change (for all status changes) so the RSS item appears new on any metadata change, including Draft to Last Call status changes. 

eipcontentfeed.html contains all the logic for this. 

I already change the title for the Last Call feed. Looking at this again, I can do the same thing for the whole feed where title has Last Call appended.

**Comment by fulldecent** [comment]
Still insufficient.

GUID uniquely identifies a post. This is true. However, the newsworthy event ("post") is not "ERC-1155 was created" or even "ERC-1155 entered last call". The correct event is "ERC-1155 entered a last call which ends on 2019-01-04".

Then when ERC-1155 enters a second last call process (which is expected) then a new article will show up in my RSS feed, which is the correct behavior.

**Comment by bmann** [comment]
@fulldecent unfortunately not possible with a static site generator like jekyll.

Or at least, not without a completely custom Ruby plugin or other custom code. Sorry!

The behaviour of “showing up as new” is very RSS client specific. Can you tell me which reader you are using?

**Comment by bmann** [comment]
>"Showing up as new" is not implementation-specific.

@fulldecent this has been my experience in using RSS for many years. So call it UI specific if you prefer. Please, it would be helpful for me if you tell me which client you use?

I can change the permalink setting.

Do you have any other comments on how you would like this to work since it’s not possible to do it the way you want with Jekyll?

I would say this is an improvement (adding a feed for all EIPs, Last Call and other status changes highlighted by feed changes, all EIP metadata displayed in the feed) that we could commit and build from.

What do you think?

**Comment by bmann** [comment]
@fulldecent I have removed the `isPermalink` and appended status to that line item. Whenever status changes, the GUID will change.

**Comment by bmann** [comment]
@fulldecent @Arachnid should I keep working on this?

**Comment by fulldecent** [comment]
@bmann  Reviewed everything. Recommend changes at https://github.com/spadebuilders/EIPs/pull/17.

Also solves this issue:

screen shot 2019-02-05 at 10 08 40 pm

**Comment by bmann** [comment]
Merged! Should be ready to go.

**Comment by fulldecent** [comment]
🤝

#1693 is good to ship 🚢

Announced at

File: _config.yml [hunk 1]
```diff
@@ -55,3 +60,6 @@ exclude:
   - vendor/gems/
   - vendor/ruby/
   - eip-X.md
+  - ISSUE_TEMPLATE.md
```
Comment (Arachnid) [inline _config.yml]: Pretty sure this shouldn't be here twice.

File: _includes/eipfeedcontent.html [hunk 2]
```diff
@@ -0,0 +1,22 @@
+<p><strong>EIP-#{{ eip.eip }}: {{ eip.title }}</strong> is in {{ eip.status }} status. It is authored by {{ eip.author | escape }} and was originally created {{ eip.created | date: '%Y-%m-%d' }}. It is in the {{ eip.category }} category of type {{ eip.type }}.</p>
```
Comment (Arachnid) [inline _includes/eipfeedcontent.html]: Not all categories have types. Can you add an if statement so the type part is omitted if it's not relevant?

Comment (Arachnid) [inline _includes/eipfeedcontent.html]: What's the motivation behind a custom template here? Why not just include the EIP itself in the feed - or perhaps only the metadata section?

File: _layouts/default.html [hunk 3]
```diff
@@ -0,0 +1,20 @@
+<!DOCTYPE html>
```
Comment (Arachnid) [inline _layouts/default.html]: This layout doesn't seem to be referenced anywhere.

File: _includes/eipfeedcontent.html (hunk 2 above)
Comment (bmann) [reply _includes/eipfeedcontent.html]: The metadata needs to get built up in this way and so needs to be "printed". I use this with some switches in the Last Call as well, and find that this human readable header is a useful way to consume in an RSS feed.

Note: I want to do the equivalent of this "all" feed as a JSON file as well, which will just be the metadata / content for consumption by machines.

File: _layouts/default.html (hunk 3 above)
Comment (bmann) [reply _layouts/default.html]: I don't think I've added the.

The way this repo is setup, you're using / inheriting the default Jekyll minima theme. I need to include it in order to override it and have header `<link>` fields that will link to the all and last call feeds.

I'll either include this directly so it's clear or remove this for a future PR.

File: _includes/eipfeedcontent.html (hunk 2 above)
Comment (bmann) [reply _includes/eipfeedcontent.html]: 👍

**Comment by bmann** [comment]
Thanks @Arachnid -- will update PR in next couple of days.

**Comment by axic** [comment]
Would be nice to merge this, but there is a merge conflict.

**Comment by axic** [comment]
@bmann are you still pursuing this?

**Comment by bmann** [comment]
@axic no, I am no longer volunteering on Ethereum.

**Pull request closed without merging**